system/system.proto

```diff
@@ -227,6 +232,86 @@ message PingResponse {
   int32 ttl = 13;           // Remaining time to live value.
 }
 
+// A ProbeRequest describes the probe operation to perform.  The
+// destination field and exactly one of the probe_target fields
+// are required.  Any field not specified is set to a reasonable server
+// specified value.  Not all fields are supported by all vendors.
+message ProbeRequest {
+  string destination = 1;   // Proxy address to probe. required.
+  string source = 2;        // Source address to probe from.
+  oneof probe_target {
```

Comment:
Well, https://fenner.github.io/probe-clarification/draft-fenner-intarea-probe-clarification.html refers to interface name and ifindex from https://www.rfc-editor.org/rfc/rfc8343 - I'm pretty ignorant of the details of ietf netmod vs. openconfig models.  I'd be happy to put in appropriate references, though, if you can help me find them.

ifaddr is expected to be the IP address assigned to some interface (presumably a state lookup, not a config one).

remote is expected to be an entry in the ARP or ND table.

The reference is also a little tricky because these are not values on the node that the RPC is being sent to, but instead, values on the `destination` that's being specified in the RPC.

Reply:
@robshakir just to clarify what you're looking for here: do you want comments like

```
message ProbeRequest {
  string destination = 1;   // Proxy address to probe. required.
  string source = 2;        // Source address to probe from.
  oneof probe_target {
  oneof probe_target {
    // exactly one of the next 4 items must be present.
    // ifIndex to probe, if present.  Equal to the value of
    // /interfaces/interface/state/if-index for the requested
    // interface on the destination node.
    int32 ifindex = 3; 
    // ifName to probe, if present.  Equal to the value of
    // /interfaces/interface/state/name for the requested
    // interface on the destination node.
    string ifname = 4;
    // ifName to probe, if present.  Equal to the value of
    // /interfaces/interface/<... IP address assigned ...> for the requested
    // interface on the destination node.
    string ifaddr = 5;
    // Remote address to probe, if present.  Equal to the value of
    // /neighbors/something/<... ARP or ND table entry ...> for the requested
    // remote neighbor on the destination node.
    string remote = 6;
  }
```

where obviously I'd fill in the right paths for "interface IP address" and "ARP/ND table" before publishing a revision